vertical-pod-autoscaler/recommender/model/container.go

```diff
@@ -126,6 +132,33 @@ func (container *ContainerState) addMemorySample(sample *ContainerUsageSample) b
 	return true
 }
 
+// RecordOOM adds info regarding OOM event in the model as an artifical memory sample.
+func (container *ContainerState) RecordOOM(timestamp time.Time, requestedMemory ResourceAmount) error {
+	resourceAmount := float64(requestedMemory)
+	// Discard old OOM
+	if timestamp.Before(container.WindowEnd.Add(-1 * MemoryAggregationInterval)) {
+		return fmt.Errorf("OOM event will be discarded - it is too old (%v)", timestamp)
+	}
+	// If OOM is in current or next window max requested mem with last known sample.
+	if timestamp.Before(container.WindowEnd.Add(MemoryAggregationInterval)) &&
+		container.MemoryUsagePeaks.Head() != nil {
+		resourceAmount = math.Max(resourceAmount, *container.MemoryUsagePeaks.Head())
+	}
+
+	// Incresa by 100MB or 20%
+	resourceAmount = math.Max(resourceAmount+100*1024*1024, resourceAmount*OOMBumpUpRatio)
```

Comment:
Can we have the 100 as a const somewhere?

Reply:
Done.